lib/sqlalchemy/orm/persistence.py

```diff
@@ -864,23 +864,21 @@ def _emit_update_statements(
     )
 
     def update_stmt():
-        clause = sql.and_()
+        clauses = []
```

Comment:
**mike bayer** (zzzeek) wrote:

well here's exactly why I have and_() and or_() working the way *I* liked them, I see it as a plain list.  I'd like to make this sql.and_(True), but then it coerces it to a single boolean and I can't append the clauses.   So we will have to do something else.

I *really like* having empty and_() and or_() behave as empty collections.  It remains intuitive for me.   So we will make it explicit by adding a special symbol that just makes the pattern *I* like work without anyone getting confused:

   # behaves exactly as it does now, boolean_empty
   # is a public symbol but nobody has to know :)
   clauses = sql.and_(boolean_empty)


This patch illustrates:

diff --git a/lib/sqlalchemy/orm/persistence.py b/lib/sqlalchemy/orm/persistence.py
index 79e7c89c3..c0945599e 100644
--- a/lib/sqlalchemy/orm/persistence.py
+++ b/lib/sqlalchemy/orm/persistence.py
@@ -862,7 +862,7 @@ def _emit_update_statements(
     )
 
     def update_stmt():
-        clause = sql.and_()
+        clause = sql.and_(sql.boolean_empty)
 
         for col in mapper._pks_by_table[table]:
             clause.clauses.append(
diff --git a/lib/sqlalchemy/sql/__init__.py b/lib/sqlalchemy/sql/__init__.py
index e0ad8d372..0974d3b90 100644
--- a/lib/sqlalchemy/sql/__init__.py
+++ b/lib/sqlalchemy/sql/__init__.py
@@ -13,6 +13,7 @@ from .expression import any_  # noqa
 from .expression import asc  # noqa
 from .expression import between  # noqa
 from .expression import bindparam  # noqa
+from .expression import boolean_empty  # noqa
 from .expression import case  # noqa
 from .expression import cast  # noqa
 from .expression import ClauseElement  # noqa
diff --git a/lib/sqlalchemy/sql/elements.py b/lib/sqlalchemy/sql/elements.py
index c1fc581ef..fa1b3bee0 100644
--- a/lib/sqlalchemy/sql/elements.py
+++ b/lib/sqlalchemy/sql/elements.py
@@ -2011,6 +2011,11 @@ class ClauseList(ClauseElement):
             return False
 
 
+boolean_empty = util.symbol(
+    "empty", doc="""construct an empty BooleanClauseList"""
+)
+
+
 class BooleanClauseList(ClauseList, ColumnElement):
     __visit_name__ = "clauselist"
 
@@ -2028,6 +2033,7 @@ class BooleanClauseList(ClauseList, ColumnElement):
         clauses = [
             _expression_literal_as_text(clause)
             for clause in util.coerce_generator_arg(clauses)
+            if clause is not boolean_empty
         ]
         for clause in clauses:
 
diff --git a/lib/sqlalchemy/sql/expression.py b/lib/sqlalchemy/sql/expression.py
index 2c55e860a..b914c8105 100644
--- a/lib/sqlalchemy/sql/expression.py
+++ b/lib/sqlalchemy/sql/expression.py
@@ -110,6 +110,7 @@ from .elements import _select_iterables  # noqa
 from .elements import _string_or_unprintable  # noqa
 from .elements import _truncated_label  # noqa
 from .elements import between  # noqa
+from .elements import boolean_empty  # noqa
 from .elements import BinaryExpression  # noqa
 from .elements import BindParameter  # noqa
 from .elements import BooleanClauseList  # noqa